src/client/datascience/jupyter/kernels/cellExecutionQueue.ts

```diff
@@ -33,22 +40,31 @@ export class CellExecutionQueue {
     }
     constructor(
         private readonly notebookPromise: Promise<INotebook>,
-        private readonly executionFactory: CellExecutionFactory,
+        private readonly cellExecutionFactory: CellExecutionFactory,
+        private readonly codeExecutionFactory: CodeExecutionFactory,
         readonly metadata: Readonly<KernelConnectionMetadata>
     ) {}
     /**
      * Queue the cell for execution & start processing it immediately.
      */
-    public queueCell(cell: NotebookCell): void {
-        const existingCellExecution = this.queueOfCellsToExecute.find((item) => item.cell === cell);
-        if (existingCellExecution) {
-            traceCellMessage(cell, 'Use existing cell execution');
-            return;
-        }
-        const cellExecution = this.executionFactory.create(cell, this.metadata);
-        this.queueOfCellsToExecute.push(cellExecution);
+    public queueCell(cell?: NotebookCell, code?: string): void {
```

Comment:
This api shows me to pass two empty arguments.
It should instead be `code : NotebookCell | string`
This way it's impossible to pass undefined and it's obvious what's expected without having to look at the code

Reply:
I think it's better to return a promise from this method, then we won't need waitForHidden.
Else you have two methods to deal with.